### PR TITLE
github: only run the coverity workflow on our repository

### PR DESCRIPTION
**Pull request by daniloegea**
Also, get the project version dynamically using meson introspect.

## Checklist

- [ ] Runs `make check` successfully.
- [ ] Retains 100% code coverage (`make check-coverage`).
- [ ] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

File: .github/workflows/coverity.yml [hunk 1]
```diff
@@ -29,4 +30,4 @@ jobs:
           tar czf netplan.tar.gz cov-int
       - name: Upload results
         run: |
-          curl --form token=${{ secrets.COVERITY_TOKEN }} --form email=${{ secrets.COVERITY_EMAIL }} --form file=@netplan.tar.gz --form version="0.106" --form description="Coverity scan" https://scan.coverity.com/builds?project=Netplan
+          curl --form token=${{ secrets.COVERITY_TOKEN }} --form email=${{ secrets.COVERITY_EMAIL }} --form file=@netplan.tar.gz --form version="$(meson introspect coveritybuild --projectinfo | jq -r .version)" --form description="Coverity scan" https://scan.coverity.com/builds?project=Netplan
```
Comment (slyon) [inline .github/workflows/coverity.yml:33]: thought (non-blocking): I wonder if we can get a more specific version for those coverity scans, using the git revision?

Something along the lines of this?
```bash
TAG=$(git describe --tags $(git rev-list --tags --max-count=1))  # find latest (stable) tag
REV=$(git rev-parse --short HEAD)  # get current git revision
VER="$TAG+git~$REV"
```

**Review by slyon — APPROVED** [review]
LGTM!
One little inline thought for consideration.

File: .github/workflows/coverity.yml (hunk 1 above)
Comment (daniloegea) [reply .github/workflows/coverity.yml:33]: sounds like a good idea

**Comment by slyon** [comment]
> `E: Package 'linux-modules-extra-5.15.0-1042-azure' has no installation candidate`
> 
> Apparently some old kernel versions were purged again...

Yeah.. I think the CI failures are related to https://ubuntu-archive-team.ubuntu.com/nbs.html cleaning. Should resolve itself eventually, once the CI runners are upgraded to a newer kernel.

**Pull request merged**